systems/plants/RigidBodyManipulator.cpp

```diff
@@ -1088,59 +1107,71 @@ void RigidBodyManipulator::doKinematicsNew(const MatrixBase<DerivedQ>& q, const
       }
 
       if (v.rows() > 0) {
-        // twist
-        auto v_body = v.middleRows(body.velocity_num_start, joint.getNumVelocities());
-        joint_twist.value().noalias() = body.J * v_body;
-        body.twist = body.parent->twist;
-        body.twist += joint_twist.value();
-
-        if (compute_gradients) {
-          // dtwistdq
-          joint_twist.gradient().value() = matGradMult(body.dJdq, v_body);
-          body.dtwistdq = body.parent->dtwistdq + joint_twist.gradient().value();
-        }
-
-        if (compute_JdotV) {
-          // Sdotv
-          auto dSdotVdqi = compute_gradients ? &body.dSdotVdqi : nullptr;
-          auto dSdotVdvi = compute_gradients ? &body.dSdotVdvi : nullptr;
-          joint.motionSubspaceDotTimesV(q_body, v_body, body.SdotV, dSdotVdqi, dSdotVdvi);
-
-          // Jdotv
-          auto joint_accel = crossSpatialMotion(body.twist, joint_twist.value());
-          joint_accel += transformSpatialMotion(body.T_new, body.SdotV);
-          body.JdotV = body.parent->JdotV + joint_accel;
+        if (joint.getNumVelocities()==0) { // for fixed joints
```

Comment:
the diff did a horrible job on this one.  I've added this if() case, but the code in the else (which was always run before) is unchanged.